commands.go

```diff
@@ -59,3 +63,77 @@ func (c *deployCommand) Func() func(*bot.Cmd) (string, error) {
 		return fmt.Sprintf(cmdResponse, output), nil
 	}
 }
+
+type resetCommand struct {
+	lock sync.Locker
+}
+
+func NewResetCommand() Command {
+	return &resetCommand{
+		lock: &sync.Mutex{},
+	}
+}
+
+func (c *resetCommand) Func() func(*bot.Cmd) (string, error) {
+	return func(cmd *bot.Cmd) (s string, e error) {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+		if len(cmd.Args) != 1 {
+			return fmt.Sprintf(invalidResetSyntax, strings.Join(cmd.Args, " ")), nil
+		}
+
+		app := cmd.Args[0]
+		output, err := c.executeSequence(app)
+		if err != nil {
+			return "", err
+		}
+
+		if isNotFound(output) {
+			return fmt.Sprintf(appNotFound, app), nil
+		}
+
+		return fmt.Sprintf(cmdResponse, output), nil
+	}
+}
+
+func (c *resetCommand) executeSequence(app string) (string, error) {
+	filename := "/tmp/st.yaml"
+	pvcDeleteCmd := "kubectl delete pvc -l app=%s"
+	statefulSetStoreCmd := "kubectl get statefulsets.apps %s -o=yaml"
+	statefulSetDeleteCmd := "kubectl delete -f /tmp/st.yaml"
+	statefulSetApplyCmd := "kubectl apply -f /tmp/st.yaml"
+
+	defer func() {
+		_ = os.Remove(filename)
+	}()
+
+	output := make([]string, 0)
+
+	statefulSetYaml := execute(fmt.Sprintf(statefulSetStoreCmd, app))
+	if isNotFound(statefulSetYaml) {
+		return statefulSetYaml, nil
+	}
+
+	f, err := os.Create(filename)
```

Comment:
pp: this is alright but you can also use https://golang.org/pkg/io/ioutil/#TempFile

Reply:
yeah, I actually prefer this method as Create overwrites and needs less arguments == more straightforward.